hooks/tk-multi-publish2/houdini/collector.py

```diff
@@ -0,0 +1,424 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import os
+import hou
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+# A dict of dicts organized by category, type and output file parm
+_HOUDINI_OUTPUTS = {
+    # rops
+    hou.ropNodeTypeCategory(): {
+        "alembic": "filename",  # alembic cache
+        "comp": "copoutput",  # composite
+        "ifd": "vm_picture",  # mantra render node
+        "opengl": "picture",  # opengl render
+        "wren": "wr_picture",  # wren wireframe
+        "geometry": "sopoutput",  # geometry node
+    },
+}
+
+
+class DDIHoudiniSessionCollector(HookBaseClass):
+
+    def process_current_session(self, settings, parent_item):
+        """
+        Analyzes the current Houdini session and parents a subtree of items
+        under the parent_item passed in.
+
+        :param dict settings: Configured settings for this collector
+        :param parent_item: Root item instance
+        """
+        items = []
+
+        # create an item representing the current houdini session
+        session_item = self.collect_current_houdini_session(settings, parent_item)
+        items.append(session_item)
+        # remember if we collect any alembic/mantra nodes
+        self._alembic_nodes_collected = False
+        self._mantra_nodes_collected = False
+        self._geometry_nodes_collected = False
+
+        # methods to collect tk alembic/mantra/geometry nodes if the app is installed
+        items.extend(self.collect_tk_alembicnodes(settings, session_item))
+        items.extend(self.collect_tk_mantranodes(settings, session_item))
+        items.extend(self.collect_tk_geometrynodes(settings, session_item))
+
+        # collect other, non-toolkit outputs to present for publishing
+        items.extend(self.collect_node_outputs(settings, session_item))
+
+        # if we have work path templates, collect matching files to publish
+        for work_template in settings["Work File Templates"].value:
+            items.extend(self.collect_work_files(settings, session_item, work_template))
+
+        # Return the list of items
+        return items
+
+    def collect_current_houdini_session(self, settings, parent_item):
```

Comment:
Unchanged method. Let it be inherited from parent?

Another reason for letting that happen is, if there is a new release from shotgunsoftware and we are merging those, then we just have to update the methods in sgtk_apps, and don't have to update any unchanged methods here (in sgtk_config).

Similarly, check all other functions (alembicnode, mantranode, work_files) and remove from this file if there are no changes.